### ISSUE TITLE
Non-convergence during the pre-training stage

### ISSUE THREAD

**Issue by 55Dupup** [clip-rt/clip-rt#4]
Dear author, when I was replicating, I found that the contrast loss was basically stable at around 0.3 to 0.5 in the first stage and could not decrease. Is this situation normal? What could be the reason for this?
Also, I want to know if the discrete action ultimately mapped to is the action that is ultimately executed by the robotic arm?
I would be most grateful if you could answer it. 😸

**Comment by gicheonkang** [comment]
@55Dupup Thank you for asking. 

A1: What did you mean by "the first stage"? If you're referring to the pre-training stage, that seems reasonable. This is because each epoch is constructed by sampling 1M transition data points from a large pool of OXE data--meaning that each epoch is different. In our experience, training over 10 epochs seems to be sufficient for pre-training.

A2: Yes, each language motion is mapped to the low-level robotic action, end-effector poses, which are further converted into lower-level joint angles using inverse kinematics (IK).

Hope this helps!
